reflex/components/core/match.py

```diff
@@ -230,28 +228,22 @@ def _create_match_cond_var_or_component(
 
         # Validate the match cases (as well as the default case) to have Var return types.
         if any(
-            case for case in match_cases if not types._isinstance(case[-1], BaseVar)
-        ) or not types._isinstance(default, BaseVar):
+            case for case in match_cases if not types._isinstance(case[-1], Var)
+        ) or not types._isinstance(default, Var):
             raise ValueError("Return types of match cases should be Vars.")
 
-        # match cases and default should all be Vars at this point.
-        # Retrieve var data of every var in the match cases and default.
-        var_data = [
-            *[el._var_data for case in match_cases for el in case],
-            default._var_data,  # type: ignore
-        ]
-
-        return match_cond_var._replace(
+        return ImmutableVar(
```

Comment:
created https://linear.app/reflex-dev/issue/REF-3532/create-matchvar-to-handle-rxmatch to eventually replace `format_match` with an actual Var